cmd/oras/internal/option/target.go

```diff
@@ -102,14 +102,12 @@ func (opts *Target) Parse() error {
 		if len(opts.headerFlags) != 0 {
 			return errors.New("custom header flags cannot be used on an OCI image layout target")
 		}
-		return nil
+		return opts.parseOCILayoutReference()
 	default:
 		opts.Type = TargetTypeRemote
-		if ref, err := registry.ParseReference(opts.RawReference); err != nil {
-			return err
-		} else if ref.Registry == "" || ref.Repository == "" {
+		if _, err := registry.ParseReference(opts.RawReference); err != nil {
 			return &oerrors.Error{
-				Err:            fmt.Errorf("%q is an invalid reference", opts.RawReference),
+				Err:            fmt.Errorf("%w: %q", err, opts.RawReference),
```

Comment:
By convention, the `err` is always at the end.
```suggestion
				Err:            fmt.Errorf("%q: %w", opts.RawReference, err),
```